packages/pocket-ic/tests/tests.rs

```diff
@@ -3177,3 +3177,225 @@ fn test_invalid_specified_id() {
     let expected_err = format!("The `specified_id` {} is invalid because it belongs to the canister allocation ranges of the test environment.\\nUse a `specified_id` that matches a canister ID on the ICP mainnet and a test environment that supports canister creation with `specified_id` (e.g., PocketIC).", specified_id);
     assert!(err.contains(&expected_err));
 }
+
+#[test]
+#[should_panic(
+    expected = "Creating an HTTP gateway requires `AutoProgress` to be enabled via `initial_time`."
+)]
+fn with_http_gateway_config_but_no_auto_progress() {
+    let http_gateway_config = InstanceHttpGatewayConfig {
+        ip_addr: None,
+        port: None,
+        domains: None,
+        https_config: None,
+    };
+    let _pic = PocketIcBuilder::new()
+        .with_application_subnet()
+        .with_http_gateway(http_gateway_config)
+        .build();
+}
+
+async fn list_instances(server_url: &Url) -> Vec<String> {
```

Comment:
We already have a function `PocketIc::list_instances`, but that function does not take a server URL as argument (it tries to reuse an existing PocketIC server based on PID).